modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java

```diff
@@ -196,11 +190,11 @@ protected void evaluateDDMFormRule(
 			evaluateDDMFormRuleAction(
 				stream.collect(Collectors.joining(" AND ")));
 
-			actionsAlreadyEvaluated.add(actions.toString());
+			_actionsAlreadyEvaluated = actions;
 		}
 		else {
 			_ddmFormRuleHelper.checkFieldAffectedByAction(
-				ddmFormRule, actionsAlreadyEvaluated);
+				ddmFormRule, _actionsAlreadyEvaluated);
```

Comment:
@gabrielibson, do you think you can clone the `ddmFormRule` object, then filter its actions to be only the actions not evaluated yet (using the `_actionsAlreadyEvaluated` to perform the filter) and then call `_ddmFormRuleHelper.checkFieldAffectedByAction` passing the cloned `ddmFormRule` as a parameter? I believe it can avoid the changes in `DDMFormEvaluatorRuleHelper` class. Can you please give it a try and re-send?

Thanks.

Reply:
In addition, please squash the commits for we deliver a small and clear pull. 😉

Reply:
Thanks, @gabrielibson! 🙂